elyra/pipeline/processor_kfp.py

```diff
@@ -158,7 +158,9 @@ def process(self, pipeline):
                 else:
                     kfp_argo_compiler.Compiler().compile(pipeline_function, pipeline_path)
             except Exception as ex:
-                raise RuntimeError('Error compiling pipeline {} for engine {} at {}'.
+                if ex.__cause__:
```

Comment:
I'm curious why these conditionals were introduced when they weren't there before.

Reply:
My thinking here was that if an exception is raised in `_upload_dependencies_to_object_store` in `processor.py`, it will set `__cause__` and have a specific error message that we want display, e.g. "Connection was refused due to incorrect Object Storage credentials ...". In case of a failure in some other function (e.g. one without a try/except block), `__cause__` may not be set and we can display the catchall "Error pre-processing pipeline" message.

Reply:
I see - that makes sense - thank you.